File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    
    unordered_map<string, int> m;
    
    for (int i = 0;i<participant.size();i++) {
        m[participant[i]]++;
    }
    
    for (int i=0;i<completion.size();i++) {
        m[completion[i]]++;
    }
    
    vector<pair<string, int>> v(m.begin(), m.end());
    
    for(int i=0;i<v.size();i++) {
        if (v[i].second % 2 == 1) return v[i].first;
    }

}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {        
        HashMap<String, Integer> par = new HashMap<>();
        
        for (String p : participant) {
            par.put(p, par.getOrDefault(p, 0) + 1);
        }
        
        for (String c : completion) {
            par.put(c, par.get(c) - 1);
        }
        
        for (String key : par.keySet()) {
            if (par.get(key) > 0) return key;
        }

        return "";
    }
}